src/components/Admin/Preview/Viewer.js

```diff
@@ -0,0 +1,100 @@
+import React, {Component} from 'react';
+import firebase from 'firebase/app';
+import 'firebase/firestore';
+import ItemSelector from '../ItemSelector';
+import ImagePreview from './ImagePreview';
+import getStore from '../../../utils/honestyStore.js';
+
+export default class Viewer extends Component {
+	constructor(props) {
+		super(props);
+
+		this.state = {
+			item: 'all',
+			images: []
+		};
+
+		this.items = {
+			all: {name: 'All Items', id: 'all'},
+			unknown: {name: 'Unknown Item', id: 'unknown'}
+		};
+	}
+
+	componentDidMount() {
+		firebase.initializeApp({
+			apiKey: 'AIzaSyBVuVNKx-rx2ON0RxbfGfbGPpiymbMrxj8',
+			authDomain: 'honesty-store-kiosk.firebaseapp.com',
+			projectId: 'honesty-store-kiosk'
+		});
+		this.db = firebase.firestore();
+		this.db.settings({timestampsInSnapshots: true});
+
+		getStore().then(store => {
+			Object.assign(this.items, store);
+			this.forceUpdate();
```

Comment:
can `this.items` go in the state? would allow us to avoid using `forceUpdate()`

Reply:
There are changes made to it when you are adding items that would involve modifying state directly.

Reply:
Can you point me to where the above is happening? It'd be good if we could get this in state

Reply:
My mistake, I was thinking of a different file. I'll put it in state.